app/controllers/api/v1/accounts/microsoft/authorizations_controller.rb

```diff
@@ -0,0 +1,25 @@
+class Api::V1::Accounts::Microsoft::AuthorizationsController < Api::V1::Accounts::BaseController
+  include MicrosoftConcern
+  before_action :check_authorization
+
+  def create
+    redirect_url = microsoft_client.auth_code.authorize_url({
+                                                              redirect_uri: "#{base_url}/microsoft/callback",
+                                                              scope: 'offline_access https://outlook.office.com/IMAP.AccessAsUser.All
+                                                            https://outlook.office.com/SMTP.Send openid',
+                                                              prompt: 'consent'
+                                                            })
+    if redirect_url
+      ::Redis::Alfred.setex('current_account_id', Current.account.id)
+      render json: { success: true, url: redirect_url }
+    else
+      render json: { success: false }, status: :unprocessable_entity
+    end
+  end
```

Comment:
- would be better to redirect them microsoft/callback/uniquecode  and the `::Redis::Alfred.setex('current_account_id', Current.account.id)`  to use that unique code as the key. Otherwise multiple users trying to connect inbox at the same time would create problems.

Reply:
I tried to do that, even more, I was trying to redirect it to the `{account.id}/microsoft/callback`, but we don't get to set any code in the redirect_uri else it gives the redirect_url mismatch error, we get the `auth_code` as params after the authorization. 

I get your point, but I tried to set the `wildcard` in the URL and couldn't.

Whatever redirect_url we pass should match the redirect_uri we set in the `azure-portal` app.

Reply:
@sojan-official I was thinking of having one more step before the authorization step to add emails[as a uniqcode] and then after the authorization callback we get an 'authorized email' from users_data. We can get the account based on that users_data['email']